pkgs/build-support/trivial-builders/default.nix

```diff
@@ -471,23 +475,36 @@ rec {
   symlinkJoin =
     args_@{ name
     , paths
+    , stripPrefix ? ""
     , preferLocalBuild ? true
     , allowSubstitutes ? false
     , postBuild ? ""
+    , failOnMissing ? stripPrefix == ""
     , ...
     }:
+    # Ensure no partial paths used, it will be confusing considering
+    # most of the time symlinkJoin is used with packages.
+    assert (stripPrefix == "" || (hasPrefix "/" stripPrefix) && stripPrefix != "/");
```

Comment:
Let's use `lib.assertMsg` and turn this comment into a user-facing error message.